packages/js/src/server/integrations/unhandled_rejection_monitor.ts

```diff
@@ -5,38 +5,62 @@ import { Types } from '@honeybadger-io/core'
 export default class UnhandledRejectionMonitor {
   protected __isReporting: boolean
   protected __client: typeof Client
+  protected __listener: (reason: unknown, _promise: Promise<unknown>) => void
 
-  constructor(client: typeof Client) {
+  constructor() {
     this.__isReporting = false
+    this.__listener = this.makeListener()
+  }
+
+  setClient(client: typeof Client) {
     this.__client = client
   }
 
+  makeListener() {
+    const honeybadgerUnhandledRejectionListener = (reason: unknown, _promise: Promise<unknown>) => {
+      if (!this.__client || !this.__client.config.enableUnhandledRejection) {
+        if (!this.hasOtherUnhandledRejectionListeners() && !this.__isReporting) {
+          fatallyLogAndExit(reason as Error)
+        }
+        return
+      }
+  
+      this.__isReporting = true;
+      this.__client.notify(reason as Types.Noticeable, { component: 'unhandledRejection' }, {
```

Comment:
Unrelated to the PR itself here, but shouldn't we send more metadata to HB since we have the actual `_promise` available?

Reply:
Good question! Usually the "reason" is an error, in which case I think this should work fine. But not always! Here's an issue for us to look further into it: https://github.com/honeybadger-io/honeybadger-js/issues/1180